packages/debug-helper/modules/inc/mockers/class-nonces-runner.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * The Mocker Runner that creates mock nonces.
+ *
+ * @package Jetpack
+ */
+
+namespace Automattic\Jetpack\Debug_Helper\Mocker;
+
+require_once __DIR__ . '/class-tools.php';
+
+/**
+ * Creating the mock nonces.
+ */
+class Nonces_Runner implements Runner_Interface {
+
+	/**
+	 * Generate random nonces.
+	 *
+	 * @param int $number Number of nonces to generate.
+	 *
+	 * @return bool
+	 */
+	public function run( $number ) {
+		for ( $i = $number, $per_batch = 500; $i > 0; $i -= $per_batch ) {
+			$this->run_batch( min( $per_batch, $i ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Add a batch of mock options.
+	 *
+	 * @param int $limit How many options to add.
+	 *
+	 * @return bool
+	 */
+	private function run_batch( $limit ) {
+		global $wpdb;
+
+		if ( ! $limit ) {
+			return false;
+		}
+
+		$sql = "INSERT INTO {$wpdb->prefix}options (option_name, option_value, autoload) VALUES ";
+
+		for ( $i = 0; $i < $limit; ++$i ) {
+			list( $name, $value ) = $this->get_random_nonce();
+			$sql                 .= "('" . esc_sql( $name ) . "', '" . esc_sql( $value ) . "', 'no'), ";
+		}
+
+		// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+		$wpdb->query( substr( $sql, 0, -2 ) );
+
+		return true;
+	}
+
+	/**
+	 * Generate a random nonce
+	 *
+	 * @return array [ $option_key, $option_value ]
+	 */
+	private function get_random_nonce() {
+		// phpcs:ignore WordPress.WP.AlternativeFunctions.rand_rand
```

Comment:
Why not `wp_rand()` ? Adding so many `phpcs` exceptions might create precedents that will be hard to clear out in the future.

Reply:
`wp_rand()` is too slow, and we don't really care about the results being unpredictable.
In fact, switching to `wp_rand()` makes the whole process run 250% to 500% slower than if we were using `rand()`.
It's called multiple times for each record generated, so the difference quickly adds up if you need to insert hundreds of thousands of records.

Reply:
Do we care for the values being random? Why not use a simple decreasing counter from `time()` with some step?

Would that make it too predictable for our testing?

Reply:
The string can be predictable, as long as the string format matches.
However, by passing the timestamp into the `get_random_string()` function we would add unnecessary complexity, while `rand()` does the job just fine.